### ISSUE TITLE
Normals of a cell

### ISSUE THREAD

**Issue by rgoermer** [dealii/dealii#15025]
Considering a 2D cell with its faces, it stated [in the glossary](https://www.dealii.org/developer/doxygen/deal.II/DEALGlossary.html#GlossFaceOrientation), that the normals of faces 0 and 2 have normals pointing into the cell, whereas faces 1 and 3 have normals pointing outward. From the documentation of [GeometryInfo](https://www.dealii.org/current/doxygen/deal.II/structGeometryInfo.html), I infer the internal numbering of faces to be 
```
*       3
*    2-->--3
*    |     |
*   0^     ^1
*    |     |
*    0-->--1
*        2
```
This would then relate to the following sheme

```
      *          ^
      *          |
      *    + --------- +
      *    |           |
      *    |           |
      *    |           |
      *    |-->        |-->
      *    |           |
      *    |     ^     |
      *    |     |     |
      *    + --------- +
```
that all makes sense up until the point, where I read the documentation of [GeometryInfo<dim>::unit_normal_orientation()](https://www.dealii.org/developer/doxygen/deal.II/structGeometryInfo.html#af9b79386bf8ca6410c906c3b8c4e8c40), where it is stated, that we take the first 4 (`2*dim`) entries of `{-1,1,-1,1,-1,1,-1,1}`, where `+-1` indicates positive/negative coordinate direction of the normal vector. Taking this definition, I arrive at the following sheme for the normal vectors:
```
      *          ^
      *          |
      *    + --------- +
      *    |           |
      *    |           |
      *    |           |
      * <--|           |-->
      *    |           |
      *    |           |
      *    |           |
      *    + --------- +
      *          |
      *          v
```
Simply put, the normals of faces 0 and 2 are flipped with this definition. 

From computing jump effects along edges, I assume the latter is the correct case, since for two neighboring cells **Q1** and **Q2** at a face **F** the jump is given by [[ du/dn ]](**F**) = [du/dn](**Q1**) - [du/dn](**Q2**) = [du/dn](**Q1**) + [du/d(-n)](**Q2**). And with different signs at the face from different cells, this makes summation quite easy in coding. 
Please excuse the typesetting, I don't know how to TeX here. 

Does anyone have any insight on the matter?

**Comment by bangerth** [comment]
There is a difference between "the normal direction with regard to a face (considered in isolation)" (which corresponds to the second of the pictures above) and "the normal direction to the boundary of the cell, at a face (considered as part of the boundary of a cell)" (which corresponds to the third picture.

Does this make sense? Would you like to propose text improvements for the documentation?

**Comment by rgoermer** [comment]
That actually makes sense. 

The only confusing matter here is the link to [GeometryInfo<dim>::unit_normal_orientation()](https://www.dealii.org/developer/doxygen/deal.II/structGeometryInfo.html#af9b79386bf8ca6410c906c3b8c4e8c40), within the description of [face orientations](https://www.dealii.org/developer/doxygen/deal.II/DEALGlossary.html#GlossFaceOrientation), as they both describe the respective case. 

It might be reasonable to put this into two subsections

- Face orientation considered in isolation
- Face orientation as boundary of a cell

with respective references.

**Comment by bangerth** [comment]
Would you be interested in writing a patch?

**Comment by bangerth** [comment]
@rgoermer I came past this issue again. It's been a while, but perhaps I could still interest you into writing a patch that addresses the point you make above? :-)
